### PR TITLE
expression: implement vectorized evaluation for `builtinStrcmpSig`

### PR DESCRIPTION
**Pull request by TommyCpp**
### What problem does this PR solve? 
implement vectorized evaluation for builtinStrcmpSig , for #12103

### What is changed and how it works?
```
go test -v -benchmem -bench=BenchmarkVectorizedBuiltinStringFunc -run=BenchmarkVectorizedBuiltinStringFunc -args "builtinStrcmpSig"
goos: linux
goarch: amd64
pkg: github.com/pingcap/tidb/expression
BenchmarkVectorizedBuiltinStringFunc/builtinStrcmpSig-VecBuiltinFunc-12                    85916             13255 ns/op               0 B/op          0 allocs/op
BenchmarkVectorizedBuiltinStringFunc/builtinStrcmpSig-NonVecBuiltinFunc-12                 27453             44354 ns/op               0 B/op          0 allocs/op
BenchmarkVectorizedBuiltinStringFunc/builtinStrcmpSig-VecBuiltinFunc#01-12                 67327             18046 ns/op               0 B/op          0 allocs/op
BenchmarkVectorizedBuiltinStringFunc/builtinStrcmpSig-NonVecBuiltinFunc#01-12              28220             42774 ns/op               0 B/op          0 allocs/op
[2019/11/12 17:13:23.121 -05:00] [WARN] [builtin_string.go:268] ["unexpected `Flen` value(-1) in CONCAT's args"] ["arg's index"=0]
[2019/11/12 17:13:23.121 -05:00] [WARN] [builtin_string.go:268] ["unexpected `Flen` value(-1) in CONCAT's args"] ["arg's index"=1]
[2019/11/12 17:13:23.121 -05:00] [WARN] [builtin_string.go:268] ["unexpected `Flen` value(-1) in CONCAT's args"] ["arg's index"=2]
PASS
ok      github.com/pingcap/tidb/expression      6.025s

```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by TommyCpp** [comment]
I am a little confused that why the coverage decrease where I haven't change anything. As for the implentation of `builtinStrcmpSig`, I guess the test cover everypart except the error handling

File: expression/builtin_string_vec.go [hunk 1]
```diff
@@ -883,11 +883,45 @@ func (b *builtinRTrimSig) vecEvalString(input *chunk.Chunk, result *chunk.Column
...
+	for i := 0; i < n; i++ {
+		// if left or right is null, then set to null and return 0(which is the default value)
+		if leftBuf.IsNull(i) {
+			result.SetNull(i, true)
+			continue
+		}
+		if rightBuf.IsNull(i) {
+			result.SetNull(i, true)
+			continue
+		}
```
Comment (wshwsh12) [inline expression/builtin_string_vec.go]: I think `if result.IsNull(i) {continue}` is enough .

File: expression/builtin_string_vec.go [hunk 2]
```diff
@@ -883,11 +883,45 @@ func (b *builtinRTrimSig) vecEvalString(input *chunk.Chunk, result *chunk.Column
...
 
 func (b *builtinStrcmpSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinStrcmpSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	// read left
```
Comment (b41sh) [inline expression/builtin_string_vec.go]: This comment doesn't make sense. We can remove it.

File: expression/builtin_string_vec_test.go [hunk 3]
```diff
@@ -238,6 +238,9 @@ var vecBuiltinStringCases = map[string][]vecExprBenchCase{
 			},
 		},
 	},
+	ast.Strcmp: {
+		{retEvalType: types.ETInt, childrenTypes: []types.EvalType{types.ETString, types.ETString}},
```
Comment (qw4990) [inline expression/builtin_string_vec_test.go]: Please use `selectStringGener` otherwise almost all results are false.

Comment (TommyCpp) [reply expression/builtin_string_vec_test.go]: OK, I will add test case when two input are same.

**Pull request merged**